fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1557,24 +1579,30 @@ public Pair<Table, Column> getRefBaseTablePartitionColumn() {
             return null;
         }
         if (refBaseTableColumnOpt.isEmpty()) {
-            Expr partitionExpr = getPartitionRefTableExprs().get(0);
-            List<SlotRef> slotRefs = Lists.newArrayList();
-            partitionExpr.collect(SlotRef.class, slotRefs);
-            Preconditions.checkState(slotRefs.size() == 1);
-            SlotRef partitionSlotRef = slotRefs.get(0);
-            for (BaseTableInfo baseTableInfo : baseTableInfos) {
-                Table table = MvUtils.getTableChecked(baseTableInfo);
-                if (partitionSlotRef.getTblNameWithoutAnalyzed().getTbl().equals(table.getName())) {
-                    refBaseTableColumnOpt = Optional.of(Pair.create(table, table.getColumn(partitionSlotRef.getColumnName())));
-                    break;
+            synchronized (this) {
```

Comment:
Is the code block wrapped by synchronized(this) time-consuming? if so, suggest to modify to 
```java
            
            	 Optional<...> tentativeRefBaseTableColumnOpt = Optional.empty();
                 if (refBaseTableColumnOpt.isEmpty()) {
                    Expr partitionExpr = getPartitionRefTableExprs().get(0);
                    List<SlotRef> slotRefs = Lists.newArrayList();
                    partitionExpr.collect(SlotRef.class, slotRefs);
                    Preconditions.checkState(slotRefs.size() == 1);
                    SlotRef partitionSlotRef = slotRefs.get(0);
                    for (BaseTableInfo baseTableInfo : baseTableInfos) {
                        Table table = MvUtils.getTableChecked(baseTableInfo);
                        if (partitionSlotRef.getTblNameWithoutAnalyzed().getTbl().equals(table.getName())) {
                            tentativeRefBaseTableColumnOpt =
                                    Optional.of(Pair.create(table, table.getColumn(partitionSlotRef.getColumnName())));
                            break;
                        }
                    }
                    if (tentativeRefBaseTableColumnOpt.isEmpty()) {
                        String baseTableNames = baseTableInfos.stream()
                                .map(tableInfo -> MvUtils.getTableChecked(tableInfo).getName())
                                .collect(Collectors.joining(","));
                        throw new RuntimeException(
                                String.format("can not find partition info for mv:%s on base tables:%s", name, baseTableNames));
                    }
          	}
          	
            synchronized (this) {
                if (refBaseTableColumnOpt.isEmpty()) {
                   refBaseTableColumnOpt = tentativeRefBaseTableColumnOpt;
                   
          	}
             } 
```

Reply:
fixed, thx.